fss/src/app/children/child-list.component.ts

```diff
@@ -72,12 +87,45 @@ export class ChildList implements OnInit, OnChanges {
     }
 
     filterChildren(event: any = null) {
+        console.log("Current Search Category: " + this.currentSearchCategory.viewValue);
         if (event) {
-            this.filteredChildren = this.allChildren.filter(Child => Child.child_pinyin_name.toLowerCase().indexOf(event.target.value.toLowerCase()) !== -1)
+            this._filterChildrenWithString(event.target.value);
         } else {
-            this.filteredChildren = this.allChildren;
+            this._filterChildrenWithString(this.searchText);
+            //this.filteredChildren = this.allChildren;
         }
     }
 
+    private _filterChildrenWithString(searchstring: string) {
+        console.log("filter Children with string called on string: " + searchstring);
```

Comment:
we should probably take the console.logs out. we can do it in a later commit though